test/functional/legacy/memory_usage_spec.lua

```diff
@@ -136,21 +137,19 @@ describe('memory usage', function()
   it('function capture lvars', function()
     local pid = eval('getpid()')
     local before = monitor_memory_usage(pid)
-    local fname = source([[
-      if !exists('s:defined_func')
-        func s:f()
-          let x = l:
-        endfunc
-      endif
-      let s:defined_func = 1
+    insert([[
+      func s:f()
+        let x = l:
+      endfunc
       for _ in range(10000)
         call s:f()
       endfor
     ]])
+    command('source')
     poke_eventloop()
     local after = monitor_memory_usage(pid)
     for _ = 1, 3 do
-      feed_command('so '..fname)
+      command('source')
```

Comment:
using command() instead of feed_command() fails, does that imply a race (i.e. the test was passing accidentally)?

Reply:
i think this is the same issue i noted above: test relies on one `s:f` being defined only, but using anon source here creates a new script item with a new version of `s:f`.

ideally resourcing the same buffer should just use the same script item and ID